app/src/main/java/org/eyeseetea/malariacare/domain/entity/DateQuestion.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (c) 2017.
+ *
+ * This file is part of QA App.
+ *
+ *  Health Network QIS App is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Health Network QIS App is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Foobar.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.eyeseetea.malariacare.domain.entity;
+
+import org.eyeseetea.malariacare.database.iomodules.dhis.importer.models.EventExtended;
+import org.eyeseetea.malariacare.database.utils.PreferencesState;
+import org.eyeseetea.malariacare.domain.exception.DateFormatException;
+
+import java.text.ParseException;
+import java.util.Date;
+import java.util.Locale;
+
+public class DateQuestion {
```

Comment:
@idelcano this is not a entity, It looks more like a utils class with all methods static that we have to try to avoid. A entities are domain concepts for example value, survey, question but not exists yet. In this case the responsability to convert date formats is of adapters, remember that adapters translate information from and to domain. We still do not have very defined domain and adapters according to clean architecture but conversions with current state can be done from ConvertToSDKVisitor, ConvertFromSDKVisitor and from activity, fragment or adapter that render the date on the screen.